tests/test_downloadermiddleware_httpcompression.py

```diff
@@ -374,13 +376,15 @@ def test_process_response_head_request_no_decode_required(self):
         self.assertStatsEqual("httpcompression/response_bytes", None)
 
 
-class HttpCompressionSubclassTest(TestCase):
+class HttpCompressionSubclassTest(unittest.TestCase):
+    @inlineCallbacks
     def test_init_missing_stats(self):
         class HttpCompressionMiddlewareSubclass(HttpCompressionMiddleware):
             def __init__(self):
                 super().__init__()
 
-        crawler = get_crawler(Spider)
+        crawler = get_crawler(NoRequestsSpider)
+        yield crawler.crawl()
```

Comment:
again, should the crawler be stopped? Or is it not required, and there is no clean-up needed?

Reply:
(there are a few more places; I won't be adding comments about them, as they seem redundant)